### DISCUSSION TITLE
show IP address & status on i2c display

### DISCUSSION THREAD

**Discussion by lionlggnd** [arendst/Tasmota#18508] [Q&A]
i need to show the internal ip, current time, the relay trigger status and the switch trigger status on a i2c display.
how i can configure this in tasmota? maybe with a refresh every minute.
Thanks :)

**Comment by sfromis** [comment]
You can do such things using rules

**Reply by lionlggnd** [reply]
Can you write the rules

**Reply by barbudor** [reply]
Can you read the docs?
https://tasmota.github.io/docs/Rules/
https://tasmota.github.io/docs/Displays/

**Comment by pkkrusty** [comment]
Reading the docs is a good start to get familiar with syntax. Here's something that might work (I haven't tested it) to get you started:
```
rule1
on time#minute do backlog Status 5; Status 10; Status 11; displaytext [c1l1t] %var1%; displaytext [c1l2]%var2%; displaytext [c1l3]%var3% endon
on StatusNET#IPAddress do var1 %value% endon
on StatusSNS#Switch1 do var2 %value% endon
on StatusSTS#Power1 do var3 %value% endon
```
Status of switches/power state/time will be one minute behind, but you'll always be one minute out of date if you're refreshing once a minute. A more timely variant would be triggering on var3 state change:
```
rule1
on time#minute do backlog Status 5; Status 10; Status 11 endon
on StatusNET#IPAddress do var1 %value% endon
on StatusSNS#Switch1 do var2 %value% endon
on StatusSTS#Power1 do var3 %value% endon
on var3#state do backlog displaytext [c1l1t] %var1%; displaytext [c1l2]%var2%; displaytext [c1l3]%var3% endon
```

**Reply by pkkrusty** [reply]
Also need to think about burn-in on a OLED display, and correct commands at boot time (Displaymodel xx, Displaymode 0, Power2 1)

**Reply by lionlggnd** [reply]
I want to use on lcd display
